components/openid-token-builder/src/main/java/com/wso2telco/openidtokenbuilder/MIFEOpenIDTokenBuilder.java

```diff
@@ -459,23 +458,24 @@ private String getAccessTokenIssuedTime(String accessToken, OAuthTokenReqMessage
     public OAuthAppDO getAppInformation(OAuth2AccessTokenReqDTO tokenReqDTO)
             throws IdentityOAuth2Exception, InvalidOAuthClientException {
         OAuthAppDO oAuthAppDO=null;
-        BaseCache<String, OAuthAppDO> appInfoCache = new BaseCache<String, OAuthAppDO>(
-                "AppInfoCache"); //$NON-NLS-1$
-        if (appInfoCache != null) {
+
+        Cache<String, OAuthAppDO> appInfoCache = Caching.getCacheManager("IdentityApplicationManagementCacheManager")
+                .getCache("AppInfoCache");
+
+        if (null != appInfoCache) {
             if (log.isDebugEnabled()) {
                 log.debug("Successfully created AppInfoCache under " //$NON-NLS-1$
                         + OAuthConstants.OAUTH_CACHE_MANAGER);
             }
-            try {
-                oAuthAppDO = appInfoCache.getValueFromCache(tokenReqDTO.getClientId());
-            }catch (Exception e){
-                log.error("Error occurred while fetching application information from cache");
-            }
-            if (oAuthAppDO == null){
+
+            if(appInfoCache.containsKey(tokenReqDTO.getClientId())){
```

Comment:
I think, it's better to avoid containsKey whenever possible. As i feel, there's little difference between fetching the entry and checking for it's nullity compared to checking the existence of the key and then fetching it. get call would anyway check for the keys existence before getting the actual entry.